consensus/src/block_storage/sync_manager.rs

```diff
@@ -486,12 +486,8 @@ impl BlockRetriever {
                         e,
                     );
                     // select next peer to try
-                    if peers.is_empty() {
-                        bail!(
-                            "Failed to fetch block {} in {} attempts: no more peers available",
-                            block_id,
-                            attempt
-                        );
+                    if peers.is_empty() || attempt >= 5 {
```

Comment:
Hmm do we need to distinguish between case when we attempted 5 times, because we want to retrieve large number of blocks which is greater than 5 * MAX_BLOCKS_PER_REQUEST? In other words, limiting the retry only makes sense in case we are failing to fetch blocks - if we are succeeding, then we should continue? 

Also, slightly orthogonal but how would a node recover once it exhausts the retry?

Reply:
hmm, good point on the first one, 5 * MAX exceed our backpressure limit but we should fix it anyway.

it'll give up this time, most of the time it'll do a state sync, if it's the last honest node, it'll retry on next message (with randomness in pick_peer)

Reply:
updated

Reply:
Let's define a static constant for `MAX_FAILED_BLOCK_RETRIEVAL` instead of hardcoding here?